src/main.ts

```diff
@@ -0,0 +1,148 @@
+import { readFile } from 'fs/promises';
+
+import {
+  getInput,
+  info,
+  setFailed,
+  setOutput,
+} from '@actions/core';
+import { Octokit } from 'octokit';
+import { WebhookPayload } from '@actions/github/lib/interfaces';
+
+import { NonNullableProperties } from './utils/non-nullable-propeties';
+
+interface GithubEnv {
+  readonly token: string | null;
+  readonly repositoryPath: string | null;
+  readonly eventPath: string | null;
+}
+
+enum ReviewState {
+  Approved = 'APPROVED',
+  ChangesRequested = 'CHANGES_REQUESTED',
+  Pending = 'PENDING',
+}
+
+enum ActionInput {
+  RequiredApprovesAmount = 'requiredApprovesAmount',
+}
+
+enum ActionOutput {
+  IsApproved = 'isApproved',
+}
+
+function isExistGithubEnvironmentVariables(
+  githubEnv: GithubEnv,
+): githubEnv is NonNullableProperties<
+  GithubEnv,
+  'token' | 'eventPath' | 'repositoryPath'
+> {
+  const { token, repositoryPath, eventPath } = githubEnv;
+  return token !== null && repositoryPath !== null && eventPath !== null;
+}
+
+async function getGithubPayload(path: string): Promise<WebhookPayload> {
+  const raw = await readFile(path, { encoding: 'utf-8' });
+  const payload = JSON.parse(raw) as WebhookPayload;
+  return payload;
+}
+
+function isAcceptableAction(action: string | undefined): boolean {
+  return action === 'submitted' || action === 'review_requested';
+}
+
+function getRequiredApprovesAmount(): number {
+  const DEFAULT_AMOUNT = 1;
+  const approvesAmountFromConfig = getInput(ActionInput.RequiredApprovesAmount);
+
+  if (/\d{1,2}/.test(approvesAmountFromConfig)) {
```

Comment:
This looks more "canonical" in this case. `+` matches one or more consecutive characters
```suggestion
  if (/\d+/.test(approvesAmountFromConfig)) {
```

Reply:
Also, I'm not even sure this check is needed. We could simply get the value and check it via `isNaN(n)`